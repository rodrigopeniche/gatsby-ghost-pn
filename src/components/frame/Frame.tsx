import * as React from "react"
import Main from "./Main"
import Wrapper from "./Wrapper"
import Content from "./Content"

const Frame = ({
    children,
}: {
    children: React.ReactNode | React.ReactNode[]
}) => {
    return (
        <Main>
            <Wrapper>
                <Content>{children}</Content>
            </Wrapper>
        </Main>
    )
}

export default Frame
