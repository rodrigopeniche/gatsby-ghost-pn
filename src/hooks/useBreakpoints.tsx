import * as React from "react"
import { useState, useEffect, createContext, useContext } from "react"

const defaultValue = {}

const BreakpointContext = createContext(defaultValue)

const BreakpointProvider = ({
    children,
    queries,
}: {
    children: React.ReactNode | React.ReactNode[]
    queries: { [key: string]: string }
}) => {
    const [queryMatch, setQueryMatch] = useState({})

    useEffect(() => {
        const mediaQueryLists: { [key: string]: any } = {}
        const keys = Object.keys(queries)
        let isAttached = false

        const handleQueryListener = () => {
            const updatedMatches = keys.reduce(
                (acc: { [key: string]: any }, media) => {
                    acc[media] = !!(
                        mediaQueryLists[media] && mediaQueryLists[media].matches
                    )
                    return acc
                },
                {}
            )
            setQueryMatch(updatedMatches)
        }

        if (window && window.matchMedia) {
            const matches: { [key: string]: any } = {}
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media] = window.matchMedia(queries[media])
                    matches[media] = mediaQueryLists[media].matches
                } else {
                    matches[media] = false
                }
            })
            setQueryMatch(matches)
            isAttached = true
            keys.forEach(media => {
                if (typeof queries[media] === "string") {
                    mediaQueryLists[media].addListener(handleQueryListener)
                }
            })
        }

        return () => {
            if (isAttached) {
                keys.forEach(media => {
                    if (typeof queries[media] === "string") {
                        mediaQueryLists[media].removeListener(
                            handleQueryListener
                        )
                    }
                })
            }
        }
    }, [queries])

    return (
        <BreakpointContext.Provider value={queryMatch}>
            {children}
        </BreakpointContext.Provider>
    )
}

function useBreakpoint(): { [key: string]: boolean } {
    const context = useContext(BreakpointContext)
    if (context === defaultValue) {
        throw new Error("useBreakpoint must be used within BreakpointProvider")
    }
    return context
}
export { useBreakpoint, BreakpointProvider }
