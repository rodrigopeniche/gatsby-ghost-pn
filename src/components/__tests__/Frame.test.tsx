import * as React from "react"
import * as renderer from "react-test-renderer"
import Frame from "../frame/Frame"
import { BreakpointProvider } from "../../hooks/useBreakpoints"

describe("Frame", () => {
    it("renders correctly", () => {
        const Children = () => <div>children</div>
        const tree = renderer.create(
            <BreakpointProvider queries={{}}>
                <Frame>
                    <Children />
                </Frame>
            </BreakpointProvider>
        )
        expect(tree.toJSON()).toMatchSnapshot()
        expect(tree.root.findByProps({ children: "children" })).toEqual(
            expect.anything()
        )
    })
})
