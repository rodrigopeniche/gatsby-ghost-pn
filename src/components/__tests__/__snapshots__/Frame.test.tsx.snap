// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frame renders correctly 1`] = `
<div
  style={
    Object {
      "background": "rgb(218,221,225)",
      "paddingTop": "60px",
    }
  }
>
  <div
    style={
      Object {
        "margin": "0 auto",
        "width": "960px",
      }
    }
  >
    <div
      style={
        Object {
          "background": "#FFF",
          "borderRadius": "7px 7px 0 0",
          "display": "flex",
          "padding": "5px",
        }
      }
    >
      <div>
        children
      </div>
    </div>
  </div>
</div>
`;
